google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/util/AppEngineUtil.java

```diff
@@ -163,13 +159,61 @@ private static boolean isFlexCompat(@NotNull Project project,
         if (root != null) {
           XmlTag vmTag = root.findFirstSubTag("vm");
           if (vmTag != null) {
-            isVmTrue = Boolean.parseBoolean(vmTag.getValue().getTrimmedText());
+            return vmTag;
+          } else {
+            return root.findFirstSubTag("env");
           }
         }
       }
     }
 
-    return isVmTrue;
+    return null;
+  }
+
+  /**
+   * Determines if a project is set up like an App Engine standard project but is configured
+   * in 'compatibility' mode. This indicates that the project runs in the flexible environment.
+   *
+   * <p>A flex compat project has an appengine-web.xml with either:
+   * {@code
+   * <vm>true</vm>
+   * <env>flex</env>
+   * }
+   */
+  public static boolean isFlexCompat(@NotNull Project project, @NotNull DeploymentSource source) {
+    return isFlexCompat(project, getArtifact(source));
```

Comment:
I think this can be addressed as part of https://github.com/GoogleCloudPlatform/gcloud-intellij/issues/731. Currently it does fit into the existing pattern. But I agree it should be addressed soon.